cupy/_core/internal.pyx

```diff
@@ -53,10 +54,18 @@ cpdef inline tuple get_size(object size):
         )
         return ()
     if cpython.PySequence_Check(size):
-        return tuple(size)
-    if isinstance(size, int):
-        return size,
-    raise ValueError('size should be None, collections.abc.Sequence, or int')
+        try:
+            return tuple(size)
+        except TypeError:
+            # A numpy.ndarray unconditionally succeeds in PySequence_Check as
+            # it implements __getitem__, but zero-dim one is an unsized object
+            # and fails to make a tuple from it.
+            pass
+    try:
+        return operator.index(size),
+    except TypeError as e:
+        raise ValueError(
```

Comment:
Should we also fix to just propagate `TypeError` instead of raising a `ValueError`?
```Python
>>> import numpy
>>> numpy.zeros('a')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object cannot be interpreted as an integer
```

Reply:
I agree to propagate `TypeError` which `operator.index` raises.